src/PackageManagement/BuildIntegration/BuildIntegratedRestoreUtility.cs

```diff
@@ -165,7 +166,10 @@ public static IReadOnlyList<PackageIdentity> GetRemovedPackages(LockFile origina
                 var closure = await project.GetProjectReferenceClosureAsync();
                 var files = closure.Select(reference => reference.PackageSpecPath).ToList();
 
-                var projectInfo = new BuildIntegratedProjectCacheEntry(project.JsonConfigPath, files);
+                var projectInfo = new BuildIntegratedProjectCacheEntry(
+                    project.JsonConfigPath,
+                    files,
+                    project.PackageSpec.RuntimeGraph.Supports.Keys.OrderBy(p => p, StringComparer.Ordinal).ToArray());
```

Comment:
Based on https://github.com/NuGet/NuGet.PackageManagement/pull/31/files#diff-def53c1d38f681d1d920797beb2d7b8fL233 `RuntimeGraph.Supports` is never null. We always create an empty one.